docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,530 @@
+not necessarily as specified above, and the selectors do not have to
+start with dot.
+
+Full json path selectors will use existing `type ObjectFieldSelector`
+to extend the current implementation for resources requests and limits.
+
+```
+// ObjectFieldSelector selects an APIVersioned field of an object.
+type ObjectFieldSelector struct {
+        // Required: Version of the schema the FieldPath is written in terms of.
+        // If no value is specified, it will be defaulted to the APIVersion of the
+        // enclosing object.
+        APIVersion string `json:"apiVersion"`
+        // Required: Path of the field to select in the specified API version
+        FieldPath string `json:"fieldPath"`
+}
+```
+
+#### Examples
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox command: [ "/bin/sh","-c", "env" ]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      env:
+        - name: CPU_LIMIT
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.containers[?(@.name=="container-name")].resources.limits.cpu
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox command: ["sh", "-c", "while true; do if [[ -e /etc/labels ]]; then cat /etc/labels; fi; if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi;sleep 5; done"]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "cpu_limit"
+            fieldRef:
+              fieldPath: spec.containers[?(@.name=="container-name")].resources.limits.cpu
+```
+
+### API with partial json path selectors
+
+Partial json path selectors specify paths to resources limits and requests
+relative to the container spec.
+
+#### Environment variables
+
+This table shows how selectors can be used for various requests and
+limits to be exposed as environment variables.
+
+| Name | Selectors |
+| -------------------- | -------------------|
+| CPU_LIMIT | resources.limits.cpu |
+| MEMORY_LIMIT |resources.limits.memory |
+| CPU_REQUEST | resources.requests.cpu |
+|MEMORY_REQUEST | resources.requests.memory |
+
+Since environment variables are container scoped, so there is no need
+to specify container name as part of the partial selectors as they are
+relative to container spec.
+
+#### Volume plugin
+
+| Path | Selectors |
+| -------------------- | -------------------|
+| container_name/cpu_limit | resources.limits.cpu |
```

Comment:
Although my point is that a path with container name in it for container resources like /etc/test-container/foo looks more obvious and could be less confusing when exposing limits from multiple containers like /test-container1/foo and /test-container2/foo instead of /foo1 and /foo2. Yes it looses the flexibility that it can not be just /foo. But it can be /test-container/foo.